sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,239 @@
+from sympy import Symbol
+from sympy.physics.vector import ReferenceFrame, Point
+from sympy.physics.mechanics import inertia, RigidBody, KanesMethod, \
+    dynamicsymbols
+
+from pydy.bodies import Body
+from pydy.joints import PinJoint, SlidingJoint, CylindricalJoint
+from pydy.joints_method import JointsMethod
+
+
+class TestPinJoint():
+    def setup(self):
+        parent = Body('parent')
+        child = Body('child')
+
+        gravity = Symbol('gravity')
+        # add gravity force in y direction at the center of mass of child.
+        child.add_force((0, child.get_mass() * gravity, 0), child.get_masscenter())
+
+        self.pin_joint = PinJoint('pin_joint', parent, child, parent_point_pos=(1, 1, 1),
+                             child_point_pos=(1, 1, 1))
+
+        JM = JointsMethod([self.pin_joint], parent)
+        self.eom = JM.rhs()
+
+    def test_simple_equations_of_motion(self):
+        # parent
+        p_name = "parent"
+        p_frame = ReferenceFrame(p_name + '_frame')
+        p_masscenter = Point(p_name + '_masscenter')
+        p_masscenter.set_vel(p_frame, 0)
+        p_mass = Symbol(p_name + '_mass')
+        p_inertia = (inertia(p_frame, 1, 1, 1), p_masscenter)
+        parent = RigidBody(p_name, p_masscenter, p_frame, p_mass, p_inertia)
+        # child
+        c_name = "child"
+        c_frame = ReferenceFrame(c_name + '_frame')
+        c_masscenter = Point(c_name + '_masscenter')
+        c_masscenter.set_vel(c_frame, 0)
+        c_mass = Symbol(c_name + '_mass')
+        c_inertia = (inertia(c_frame, 1, 1, 1), c_masscenter)
+        child = RigidBody(c_name, c_masscenter, c_frame, c_mass, c_inertia)
+
+        # pinjoint
+        theta = dynamicsymbols('theta')
+        thetad = dynamicsymbols('theta', 1)
+        omega = dynamicsymbols('omega')
+
+        p_axis = p_frame.x
+        c_axis = c_frame.x
+
+        c_frame.orient(p_frame, 'Axis', [theta, p_axis])
+        c_frame.set_ang_vel(p_frame, omega * p_axis)
+
+        p_joint_point = p_masscenter.locatenew(
+            p_name + '_parent_joint',
+            p_frame.x + p_frame.y + p_frame.z)
+
+        c_joint_point = c_masscenter.locatenew(
+            c_name + '_child_joint',
+            c_frame.x + c_frame.y + c_frame.z)
+
+        c_joint_point.set_pos(p_joint_point, 0)
+        c_masscenter.v2pt_theory(p_masscenter, p_frame, c_frame)
```

Comment:
I do not think this is correct. The mass centers of the two bodies are not fixed in the same frame, so the v2pt_theory does not apply here. I think you can do this though:

```python
p_joint_point.v2pt_theory(p_masscenter, newtonian_frame, p_frame)
c_joint_point.sel_vel(p_frame, 0)
c_masscenter.v2pt_theory(c_joint_point, newtonian_frame, c_frame)
```

The above will generate the velocity of the c_masscenter wrt to the p_masscenter in the newtonian frame. But an issues here is that you don't have a newtonian frame. Ultimately you need the velocities defined in the ground frame of the tree of joints. I guess the best you can do is this for a single joint:

```python
p_joint_point.v2pt_theory(p_masscenter, p_frame, p_frame)  # this will be 0
c_joint_point.sel_vel(p_frame, 0)
c_masscenter.v2pt_theory(c_joint_point, p_frame, c_frame)
```